idaes/models/properties/modular_properties/phase_equil/smooth_VLE_2.py

```diff
@@ -0,0 +1,339 @@
+    identify_VL_component_list,
+)
+from idaes.models.properties.modular_properties.base.utility import (
+    estimate_Tbub,
+    estimate_Tdew,
+)
+from idaes.models.properties.modular_properties.eos.ceos import (
+    calculate_equilibrium_cubic_coefficients,
+)
+import idaes.core.util.scaling as iscale
+
+
+# Small value for initializing slack variables
+EPS_INIT = 1e-4
+
+
+# -----------------------------------------------------------------------------
+class SmoothVLE2:
+    """
+    Improved Vapor-Liquid Equilibrium complementarity formulation for Cubic Equations of State.
+    """
+
+    @staticmethod
+    def phase_equil(b, phase_pair):
+        """
+        Method for constructing phase equilibrium variables and constraints
+        """
+        # This method is called via StateBlock.build, thus does not need clean-up
+        # try/except statements
+        suffix = "_" + phase_pair[0] + "_" + phase_pair[1]
+
+        # Smooth VLE assumes a liquid and a vapor phase, so validate this
+        (
+            l_phase,
+            v_phase,
+            _,
+            _,
+            _,
+            _,
+        ) = identify_VL_component_list(b, phase_pair)
+
+        if l_phase is None or v_phase is None:
+            raise ConfigurationError(
+                f"{b.params.name} - Generic Property Package phase pair {phase_pair[0]}-{phase_pair[1]} "
+                "was set to use Smooth VLE formulation, however this is not a vapor-liquid pair."
+            )
+
+        try:
+            lobj = b.params.get_phase(l_phase)
+            ctype = lobj._cubic_type  # pylint: disable=protected-access
+            cname = lobj.config.equation_of_state_options["type"].name
+            vobj = b.params.get_phase(v_phase)
+
+            if (
+                ctype != vobj._cubic_type  # pylint: disable=protected-access
+                or lobj.config.equation_of_state_options
+                != vobj.config.equation_of_state_options
+            ):
+                raise ConfigurationError(
+                    f"{b.params.name} - SmoothVLE2 formulation requires that both phases use the same "
+                    "type of cubic equation of state."
+                )
+        except AttributeError:
+            raise ConfigurationError(
+                f"{b.params.name} - SmoothVLE2 formulation only supports cubic equations of state."
+            )
+
+        # Definition of equilibrium temperature for smooth VLE
+        uom = b.params.get_metadata().default_units
+        t_units = uom.TEMPERATURE
+        f_units = uom.AMOUNT / uom.TIME
+
+        vl_phase_set = Set(initialize=[phase_pair[0], phase_pair[1]])
+        b.add_component("_vle_set" + suffix, vl_phase_set)
+
+        s = Var(
+            vl_phase_set,
+            initialize=EPS_INIT,
+            bounds=(0, None),
+            doc="Slack variable for equilibrium temperature",
+            units=t_units,
+        )
```

Comment:
Do we have any sort of scaling for these slack variables, or is the intention to let the autoscaler do it? This is a case where the results from the autoscaler is going to be highly dependent on initialization, and since the slack variable will frequently be near zero, you may not get any scaling at all.